src/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs

```diff
@@ -167,29 +167,49 @@ internal static Assembly LoadGeneratedAssembly(Type type, string defaultNamespac
                 name.Name = serializerName;
                 name.CodeBase = null;
                 name.CultureInfo = CultureInfo.InvariantCulture;
-                string serializerPath = Path.Combine(Path.GetDirectoryName(type.Assembly.Location), serializerName + ".dll");
-                if (!File.Exists(serializerPath))
+
+                string serializerPath = null;
+
+                try
+                {
+                    serializerPath = Path.Combine(Path.GetDirectoryName(type.Assembly.Location), serializerName + ".dll");
+                }
+                catch
                 {
-                    serializerPath = Path.Combine(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location), serializerName + ".dll");
                 }
 
                 try
                 {
-                    serializer = Assembly.LoadFile(serializerPath);
+                    if (string.IsNullOrEmpty(serializerPath) || !File.Exists(serializerPath))
+                    {
+                        serializerPath = Path.Combine(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location), serializerName + ".dll");
```

Comment:
We probably should do the null check of `Assembly.GetEntryAssembly().Location` too.